src/Processors/Formats/Impl/HashOutputFormat.cpp

```diff
@@ -0,0 +1,70 @@
+#include<Processors/Formats/Impl/HashOutputFormat.h>
+#include <Formats/FormatFactory.h>
+#include <Common/HashTable/Hash.h>
+#include <Processors/Formats/IOutputFormat.h>
+#include <Columns/IColumn.h>
+#include <Processors/Port.h>
+#include <fmt/format.h>
+
+namespace DB
+{
+
+HashOutputFormat::HashOutputFormat(const Block & header, WriteBuffer & buf)
+    : IOutputFormat(header, buf), hashing_buffer(buf)
+{
+
+}
+
+String HashOutputFormat::getName() const
+{
+    return "HashOutputFormat";
+}
+
+String HashOutputFormat::getHash()
+{
+    const auto h = hashing_buffer.getHash();
+    std::string raw = fmt::format("{:016x}{:016x}", h.high64, h.low64);
+    size_t pos = raw.find_first_not_of('0');
+    if (pos == std::string::npos)
+        pos = raw.size() - 1;
+    return String(raw.substr(pos));
+}
+
+void HashOutputFormat::consume(Chunk chunk)
+{
+    Arena arena;
+    const char * begin = nullptr;
+
+    for (const auto & column : chunk.getColumns())
+    {
+        for (size_t i = 0; i < column->size(); ++i)
+        {
+            StringRef ref = column->serializeValueIntoArena(i, arena, begin);
+            hashing_buffer.write(ref.data, ref.size);
+        }
+    }
```

Comment:
Using `Arena` just to calculate a hash seems excessive here.
We can try to serialize data directly to `hashing_buffer` instead:

```
    const auto & columns = chunk.getColumns();
    chassert(serializations.size() == columns.size());
    for (size_t i = 0; i != columns.size(); ++i)
        serializations[i]->serializeBinaryBulk(columns[i], hashing_buffer, 0, column->size());
```
after initializing member variable `serializations` (of type `std::vector<SerializationPtr>`) in the constructor:
```
    serializations.reserve(headers.columns());
    for (const auto & header_column : header)
        serializations.push_back(header_column.type->getDefaultSerialization());
```